### PR TITLE
Add support for generation host server

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type
What kind of change does this PR introduce?
- Feature

## What is the current behavior?
The source generation is performed using an out-of-process worker, but is not kept alive. This makes incremental builds slower.

## What is the new behavior?
A new server host mode is now enabled when running on Windows or .NET core, giving the ability to retain already loaded msbuild binaries when running incremental builds.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno.SourceGeneration/tree/master/doc/.feature-template.md). (for bug fixes / features)
- [x] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/nventive/Uno.SourceGeneration/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

Internal Issue (If applicable):
https://nventive.visualstudio.com/DefaultCollection/Umbrella/_workitems/edit/143604

### CODE REVIEW

File: src/Uno.SourceGeneration.Host/Program.cs [hunk 1]
```diff
@@ -1,96 +1,37 @@
...
-using Microsoft.Build.Logging;
-using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Logging.Console;
-using Uno.SourceGeneration.Helpers;
-using Uno.SourceGeneratorTasks;
+using Uno.SourceGeneration.Host.Helpers;
+using Uno.SourceGeneration.Host.Server;
 using Uno.SourceGeneratorTasks.Helpers;
 
+[assembly: CommitHashAttribute("<developer build>")]
```
Comment (carldebilly) [inline src/Uno.SourceGeneration.Host/Program.cs]: What is that??

Comment (jeromelaban) [reply src/Uno.SourceGeneration.Host/Program.cs]: It's to make sure that the client is connected to a compatible server. Which reminds me I need to change it in the builds scripts.

File: src/Uno.SourceGeneration.Protocol/Messages/CompletedGenerationResponse.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+			// This field existed to support writing to Console.Error.  The compiler doesn't ever write to 
+			// this field or Console.Error.  This field is only kept around in order to maintain the existing
+			// protocol semantics.
+			ErrorOutput = string.Empty;
+		}
+
+		public override ResponseType Type => ResponseType.Completed;
+
+		public static CompletedGenerationResponse Create(BinaryReader reader)
```
Comment (carldebilly) [inline src/Uno.SourceGeneration.Protocol/Messages/CompletedGenerationResponse.cs]: Can't use Pipelines here? ;-)

Comment (jeromelaban) [reply src/Uno.SourceGeneration.Protocol/Messages/CompletedGenerationResponse.cs]: It's code that's coming from Roslyn, so no :)

**Pull request merged**